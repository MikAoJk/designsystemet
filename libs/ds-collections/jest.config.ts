export default {
  displayName: 'ds-collections',
  preset: '../../jest.preset.js',
  transform: {
    '^.+\\.[tj]sx?$': ['babel-jest', { presets: ['@nrwl/react/babel'] }],
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../coverage/libs/ds-buttons',
};
