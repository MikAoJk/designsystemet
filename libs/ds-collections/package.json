{
  "name": "@skatteetaten/ds-collections",
  "groupId": "no.skatteetaten.aurora",
  "version": "0.7.1",
  "homepage": "https://www.skatteetaten.no/stilogtone/designsystemet/",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Skatteetaten/designsystemet.git"
  },
  "publishConfig": {
    "registry": "https://nexus.sits.no/repository/npm-internal/"
  },
  "peerDependencies": {
    "i18next": "^23",
    "react": "^18",
    "react-i18next": "^13",
    "@skatteetaten/ds-core-utils": "*",
    "@skatteetaten/ds-icons": "*",
    "@skatteetaten/ds-buttons": "*",
    "@skatteetaten/ds-content": "*",
    "@skatteetaten/ds-typography": "*",
    "@skatteetaten/ds-progress": "*"
  },
  "scripts": {}
}
