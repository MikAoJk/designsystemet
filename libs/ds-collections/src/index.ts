export * from './Accordion/Accordion';
export * from './Accordion/Accordion.types';
export * from './AccordionItem/AccordionItem.types';
export * from './OpenClose/OpenClose';
export * from './OpenClose/OpenClose.types';
export * from './StepList/StepList';
export * from './StepList/StepList.types';
export * from './StepListStep/StepListStep.types';
export * from './Tabs/Tabs';
export * from './Tabs/Tabs.types';
