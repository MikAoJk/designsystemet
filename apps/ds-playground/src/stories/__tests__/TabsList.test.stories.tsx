import { Tabs } from '@skatteetaten/ds-collections';
import { Meta, StoryFn, StoryObj } from '@storybook/react';

const meta = {
  component: Tabs.List,
  title: 'Tester/Tabs/List',
  argTypes: {
    // Baseprops
    ref: { table: { disable: true } },
    key: { table: { disable: true } },
    className: { table: { disable: true } },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },
    // Props
    children: { table: { disable: false } },
  },
} satisfies Meta<typeof Tabs.Panel>;
export default meta;
type Story = StoryObj<typeof meta>;

const TemplateTabsList: StoryFn<typeof Tabs> = (args) => {
  return (
    <Tabs defaultValue={'tab1'}>
      <Tabs.List {...args}>
        <Tabs.Tab value={'tab1'}>{'Person'}</Tabs.Tab>
        <Tabs.Tab value={'tab2'}>{'Bedrift'}</Tabs.Tab>
      </Tabs.List>
      <Tabs.Panel value={'tab1'}>{'Tabs.Panel Panel'}</Tabs.Panel>
      <Tabs.Panel value={'tab2'}>{'Tabs.Panel Bedrift'}</Tabs.Panel>
    </Tabs>
  );
};

export const WithAttributes = {
  name: 'With Attributes (FA2-5, B1)',
  render: TemplateTabsList,
  args: {
    className: 'dummyClassname',
    lang: 'nb',
    'data-testid': '123ID',
    children: [
      <Tabs.Tab key={'k1'} value={'tab1'}>
        {'Person'}
      </Tabs.Tab>,
      <Tabs.Tab key={'k1'} value={'tab2'}>
        {'Bedrift'}
      </Tabs.Tab>,
    ],
  },
  argTypes: {
    id: { table: { disable: false } },
    className: { table: { disable: false } },
    lang: { table: { disable: false } },
    'data-testid': { table: { disable: false } },
  },
  /*   play: async ({ canvasElement, step }): Promise<void> => {
    const canvas = within(canvasElement);
    await step(
      'Autogenerert id-attributt basert pÃ¥ tab name "tab1"',
      async () => {
        const tabpanel = canvas.getByRole('tabpanel', {
          name: 'Person',
          hidden: true,
        });
        await expect(tabpanel).toHaveAttribute('id', 'ds-tab-panel-tab1');
        await expect(tabpanel).toHaveClass('dummyClassname');
        await expect(tabpanel).toHaveAttribute('lang', 'nb');
        await expect(tabpanel).toHaveAttribute('data-testid', '123ID');
      }
    );
  }, */
} satisfies Story;
