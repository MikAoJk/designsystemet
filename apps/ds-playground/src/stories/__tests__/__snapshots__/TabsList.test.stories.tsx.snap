// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/Tabs/List WithAttributes smoke-test 1`] = `
<div data-test-block="true">
  <div class>
    <div class="TabsList_tabList__vWYZt  dummyClassname"
         lang="nb"
         data-testid="123ID"
         role="tablist"
         tabindex="-1"
    >
      <button id="ds-tab-id-:r0:-tab1"
              class="TabsTab_tab__W3ulY   TabsTab_tab_active__RkJBU  "
              role="tab"
              tabindex="0"
              aria-selected="true"
              aria-controls="ds-tab-panel-:r0:-tab1"
      >
        Person
      </button>
      <button id="ds-tab-id-:r0:-tab2"
              class="TabsTab_tab__W3ulY     "
              role="tab"
              tabindex="-1"
              aria-selected="false"
              aria-controls="ds-tab-panel-:r0:-tab2"
      >
        Bedrift
      </button>
    </div>
    <div id="ds-tab-panel-:r0:-tab1"
         class="TabsPanel_panel__4lPgj TabsPanel_panel_active__pqi5b "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r0:-tab1"
    >
      Tabs.Panel Panel
    </div>
    <div id="ds-tab-panel-:r0:-tab2"
         class="TabsPanel_panel__4lPgj  "
         role="tabpanel"
         aria-labelledby="ds-tab-id-:r0:-tab2"
    >
      Tabs.Panel Bedrift
    </div>
  </div>
</div>
`;
